### PR TITLE
PackageVersion made constexpr and moved to src/package_version.hpp

### LINKED ISSUE #2052: Move version.hpp to src/package_version.hpp
Now that `PackageVersion` is in `_detail` (#2042), we should move it out from the public headers. Ideally, to `src`. Maybe, to `internal/`, but I think it should be possible to move it to `src`. If TelemetryPolicy is implemented in header and therefore would not have access to `src` header, move TelemetryPolicy's implementation to an src file.

**At the very least, rename it to `package_version.hpp`.**

### LINKED ISSUE #2053: Replace PackageVersion::ToString() with `static constexpr char[]`
See also #2052.

### PR DESCRIPTION
**Pull request by antkmsft**
Closes #2052 
Closes #2053

### CODE REVIEW

**Review by RickWinter — COMMENTED** [review]
What's the motivation for moving this to be source vs include header?

File: eng/scripts/Update-PkgVersion.ps1
Comment (vhvb1989) [inline eng/scripts/Update-PkgVersion.ps1:9]: I didn't find any string for the expected path like `inc/azure/core` or something. 
Does it means that the script will try to find `package_version.hpp` anywhere under the `sdk/ServiceDirectory` path?

**Review by vhvb1989 — APPROVED** [review]
LGTM.
I don't understand why it is bad to let the header in the /inc path inside the _detail namespace. 

Please fix the Linux/Mac and wait for some EngSys review.

File: eng/scripts/Update-PkgVersion.ps1
Comment (weshaggard) [reply eng/scripts/Update-PkgVersion.ps1:9]: See https://github.com/Azure/azure-sdk-for-cpp/pull/2166/files#diff-f78d4a430718b3caaf9acaafbcc127358ac1be36f198386ebefe14891c18870eR7 it looks under the `sdk/$ServiceDirectory/$PackageName` directory for the file.

File: sdk/storage/azure-storage-blobs/src/blob_client.cpp
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/src/blob_client.cpp:20]: extra new line

File: sdk/storage/azure-storage-blobs/src/blob_container_client.cpp
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/src/blob_container_client.cpp:17]: extra new line

**Review by Jinming-Hu — COMMENTED** [review]
Wait for storage team's approval next time.

**Pull request merged**
